akka-actor-typed/src/main/scala/akka/actor/typed/internal/BehaviorImpl.scala

```diff
@@ -115,13 +115,13 @@ import scala.reflect.ClassTag
    *
    * The returned behavior from processing messages and signals can also be
    * intercepted, e.g. to return another `Behavior`. The passed message to
-   * `afterMessage` is the message returned from `beforeMessage` (possibly
-   * different than the incoming message).
+   * `afterMessage` is the original message (not the one returned from `beforeMessage`).
```

Comment:
I think this is more powerful, so independent of the fate of this PR we should change `intercept` to this.